### PR TITLE
review-fixes

### PR DESCRIPTION
**Pull request by nir0s**

### CODE REVIEW

File: wagon.py [hunk 1]
```diff
@@ -608,7 +615,7 @@ def extract_source(source, destination):
         remove_source_after_process = True
         tmpdir = tempfile.mkdtemp()
         source = extract_source(source, tmpdir)
-    elif os.path.isdir(source):
+    elif os.path.isdir(os.path.expanduser(source)):
```
Comment (dankilman) [inline wagon.py]: this part is still buggy

File: wagon.py [hunk 2]
```diff
@@ -259,8 +266,8 @@ def install_package(package,
 
 
 def _get_downloaded_wheels(path):
-    return \
-        [filename for filename in os.listdir(path) if filename.endswith('whl')]
+    return [filename for filename in os.listdir(path)
+            if os.path.splitext(filename)[1]]
```
Comment (dankilman) [inline wagon.py]: i don't understand the new condition. if it has an extension its good enough?

File: wagon.py [hunk 3]
```diff
@@ -26,6 +25,7 @@
 import logging
 import platform
 import tempfile
+import StringIO
```
Comment (dankilman) [inline wagon.py]: to make this python3 compatible you need to try/except ImportError. the package changed in python 3

Comment (nirfuchs) [inline wagon.py]: `from io import StringIO`
will work for both Python 2 and 3

File: wagon.py (hunk 2 above)
Comment (nirfuchs) [inline wagon.py]: adding to Dan's note, i'm guessing this should have been `if os.path.splitext(filename)[1].lower() == '.whl'`

File: wagon.py (hunk 3 above)
Comment (dankilman) [inline wagon.py]: one thing to note though is that io.StringIO mandates unicode, so the code should be aware of that

Comment (nirfuchs) [inline wagon.py]: You're absolutely right :+1: 
I forgot to mention it

File: wagon.py (hunk 2 above)
Comment (nir0s) [inline wagon.py]: Yeah, that's my bad.

Need a test for that.

**Pull request merged**